File: C#/第一部分 C#语言/第四章 循环结构/第二节 while语句/while.cs
// 由bhw设计
/*
第4章 第2节 while语句

一、语法
while (条件) { 循环体 }

二、原理
1) 先判断条件，再决定是否执行循环体。
2) 适合“循环次数不固定”的问题。

三、习题
1) 输入 n，计算 n 的阶乘。
2) 持续读入整数，直到输入 0 停止并求和。

四、答案与讲解
1) while (n > 1) 做累乘。
2) while 循环中先读值，再判断终止条件。
*/
using System; // 输出库

class Program // 类定义
{
    static void Main() // 主函数
    {
        int n = 5; // 初始化 n
        long fact = 1; // 阶乘结果
        while (n > 1) // 条件为真继续
        {
            fact *= n; // 累乘
            n--; // 递减
        }
        Console.WriteLine($"fact={fact}"); // 输出结果
    } // Main结束
} // 类结束
// 由bhw设计

File: C/第一部分 C语言/第四章 循环结构/第二节 while语句/while.c
// 由bhw设计
/*
第4章 第2节 while语句
while语法:
while(条件) { 循环体 }
适合循环次数不确定的场景。

习题:
1) 输入n计算n的阶乘。
2) 连续输入整数直到输入0停止并求和。

答案与讲解:
1) while (n > 1) fact *= n--。
2) 用while读取并累计sum。
*/
#include <stdio.h> // 输入输出函数

int main(void) { // 主函数开始
    int n = 5; // 定义并初始化n
    long long fact = 1; // 定义阶乘结果变量
    while (n > 1) { // 当n大于1继续循环
        fact *= n; // 累乘当前n
        n--; // n递减, 逐步逼近终止条件
    } // while结束
    printf("fact=%lld\n", fact); // 输出阶乘结果
    return 0; // 程序正常返回
} // main结束
// 由bhw设计
